### PR TITLE
Header only xxhash submodule

### PR DESCRIPTION
**Pull request by donnadionne**
@donnadionne

### CODE REVIEW

**Comment by donnadionne** [comment]
Thanks to NIco's suggestion, this is a much easier import of xxhash submodule.

All tests (including distribution) should are passing now

File: cmake/xxhash.cmake
Comment (markdroth) [inline cmake/xxhash.cmake:19]: I assume this should be removed.

File: src/core/ext/filters/client_channel/resolver/xds/xds_resolver.cc
Comment (markdroth) [inline src/core/ext/filters/client_channel/resolver/xds/xds_resolver.cc:424]: Please remove.

File: src/xxhash/gen_build_yaml.py
Comment (markdroth) [inline src/xxhash/gen_build_yaml.py:32]: No need for `sorted()` or `glob.glob()` here, since there's just one path.

**Review by markdroth — APPROVED** [review]
LGTM, other than a few minor nits.

Please make sure Nico approves before merging.

Thanks!

**Comment by donnadionne** [comment]
Jan and Nico,
Thank you both so much for your help with this import!  Please take a look at the changes and let me know if you have more feedback.

I still have 2 small errors (objC xcode build and php artifacts), I think I have all the right changes in but I will check with Yulin and Stanley to see if I missed something...  (should be something very small)

In any case, the rest of the changes are ready for your review.

Thank you so much!

**Comment by donnadionne** [comment]
All tests now pass!  Preparing for an import and waiting for final review from Nico and Jan!

Thank you

File: cmake/xxhash.cmake
Comment (donnadionne) [reply cmake/xxhash.cmake]: removed

File: src/core/ext/filters/client_channel/resolver/xds/xds_resolver.cc
Comment (donnadionne) [reply src/core/ext/filters/client_channel/resolver/xds/xds_resolver.cc]: fixed

Comment (donnadionne) [reply src/core/ext/filters/client_channel/resolver/xds/xds_resolver.cc]: removed real code to use xxhash is coming as soon as this is submitted

File: src/xxhash/gen_build_yaml.py
Comment (donnadionne) [reply src/xxhash/gen_build_yaml.py]: fixed

File: templates/CMakeLists.txt.template
Comment (jtattermusch) [inline templates/CMakeLists.txt.template:197]: nit: typo `xxhahs`

**Review by jtattermusch — COMMENTED** [review]
Looks like several tests have failed since the PR currently has conflicts, so I wasn't able to see which tests are actually broken and which ones are not - please resolve the conflicts so that we get useful test results.

One high-level issue I don't understand is why you're going through all the hoops of to include the xxhash's cmake build into our cmake build  and setting up a new gitmodule if the only thing that you need is really the xxhash.h file.

If you can only build in the header-only mode, this is the easiest solution:
- don't create a submodule, but just include a copy of of xxhash under third_party/xxhash/xxhash.h (not clean from puristic sense, but it's just 1 file and it makes a lot of stuff so much easier).
- just make all the non-bazel builds include the xxhash.h
- add a simple xxhash.BUILD file which adds the header-only lib. (the xxhash.BUILD you currently have is very similar to it).
- Since xxhash.h is only an internal dependency, (almost) everything will just work without you needing to change anything (it's the same situation as if you added a new internal header to grpc).

The approach when you take the effort to actually support xxhash via its own cmake build in two modes (package and module), but then only use a single header from its installation doesn't really make much sense IMHO (since basically none of the stuff you took care to install will be used).

Also, I'm curious how come with the "non-header-only" approach you've tried before, things didn't compile on aarch64, but when using the "header-only" mode, things now magically work (that's nice of course but I'd like to make sure that we're not missing something and this approach will actually work on aarch64 in the future).

**Pull request closed without merging**
